developer_guides/linux_driver/third_party/keyword_detect/keyword_detect.rst

```diff
@@ -0,0 +1,140 @@
+.. _sof_driver_arch:
+
+Keyword Detection Driver implementation and user guide
+######################################################
+
+.. contents::
+   :local:
+   :depth: 1
+
+Keyword Detection pipelines
+***************************
+
+.. code-block:: none
+
+   # PCM  <-----(pipe 8)--------- host <---------------- KPB <------- Volume <--- DMIC (dmic16k)
+   #  |                                                   |
+   #  |                                                   |
+   #  |                                                   |
+   # Detector Sink <---Detector(pipe 9) <--- selector <---+
+
+We are using DAPM events to trigger the detect pipeline(pipe 9). Here the ‘Detector Sink’ is a virtual DAPM widget(visible to the driver, but not to the FW), it is used to send pipeline control IPCs(hw_params, trigger start/stop, hw_free) to firmware. These control IPCs are sent to FW in the sequence like this (1->2(2.1->2.2)->3->4->5->6(6.1->6.2)):
+
+.. code-block:: none
+
```

Comment:
Can we use rst table format? https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#simple-tables

Reply:
Just tried and looks the simple table doesn't work, maybe we are lack of numpydoc or autodoc extensions, but switching to use csv-table works fine, updating it now.